src/annotations/deprecated.cr

```diff
@@ -0,0 +1,11 @@
+# This annotations marks methods, types or constants as deprecated.
+#
+# It receives a `StringLiteral` as single argument containing a deprecation notice.
+#
+# ```cr
+# @[Deprecated("#foo has been deprecated, use #bar instead")]
```

Comment:
`top-level foo` is added as a prefix of the warning. There is no need to mention it.